code/modules/surgery/bodyparts/species_parts/genetics_bodyparts.dm

```diff
@@ -0,0 +1,222 @@
+
+	cast_on.add_filter(BUNNY_HOP, 2, drop_shadow_filter(color = "#03020781", size = 0.9))
+	var/shadow_filter = cast_on.get_filter(BUNNY_HOP)
+	var/jump_height = 8 + 16 * ((bunny_multiplier - 1) / 5)
+	var/jump_duration = 0.25 SECONDS * bunny_multiplier
+	new /obj/effect/temp_visual/mook_dust(get_turf(cast_on))
+	animate(cast_on, pixel_y = cast_on.pixel_y + jump_height, time = jump_duration / 2, easing = CIRCULAR_EASING|EASE_OUT)
+	animate(pixel_y = initial(owner.pixel_y), time = jump_duration / 2, easing = CIRCULAR_EASING|EASE_IN)
+
+	animate(shadow_filter, y = -jump_height, size = 4, time = jump_duration / 2, easing = CIRCULAR_EASING|EASE_OUT)
+	animate(y = 0, size = 0.9, time = jump_duration / 2, easing = CIRCULAR_EASING|EASE_IN)
+
+	addtimer(CALLBACK(src, PROC_REF(end_jump), cast_on), jump_duration)
+
+/datum/action/cooldown/spell/bunny_hop/proc/break_glass(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/bunny = mover
+	for(var/obj/structure/window/window in get_turf(bunny))
+		window.deconstruct(disassembled = FALSE)
+		mover.balloon_alert_to_viewers("smashed through!")
+		bunny.apply_damage(damage = rand(5,15), damagetype = BRUTE, wound_bonus = 15, bare_wound_bonus = 25, sharpness = SHARP_EDGED, attack_direction = get_dir(window, oldloc))
+		new /obj/effect/decal/cleanable/glass(get_step(bunny, bunny.dir))
+	for(var/obj/machinery/door/window/windoor in get_turf(bunny))
+		windoor.deconstruct(disassembled = FALSE)
+		mover.balloon_alert_to_viewers("smashed through!")
+		bunny.apply_damage(damage = rand(5,15), damagetype = BRUTE, wound_bonus = 15, bare_wound_bonus = 25, sharpness = SHARP_EDGED, attack_direction = get_dir(windoor, oldloc))
+		new /obj/effect/decal/cleanable/glass(get_step(bunny, bunny.dir))
+	for(var/obj/structure/grille/grille in get_turf(bunny))
+		grille.shock(bunny, 70)
+		grille.deconstruct(disassembled = FALSE)
+		mover.balloon_alert_to_viewers("smashed through!")
+		bunny.apply_damage(damage = rand(5,10), damagetype = BRUTE, wound_bonus = 5, bare_wound_bonus = 15, attack_direction = get_dir(grille, oldloc))
+		new /obj/effect/decal/cleanable/generic(get_step(bunny, bunny.dir))
+
+/datum/action/cooldown/spell/bunny_hop/proc/break_grilles(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/bunny = mover
+	for(var/obj/structure/grille/grille in get_turf(bunny))
+		grille.shock(bunny, 70)
+		grille.deconstruct(disassembled = FALSE)
+		mover.balloon_alert_to_viewers("smashed through!")
+		bunny.apply_damage(damage = rand(5,10), damagetype = BRUTE, wound_bonus = 5, bare_wound_bonus = 15, attack_direction = get_dir(grille, oldloc))
+		new /obj/effect/decal/cleanable/generic(get_step(bunny, bunny.dir))
+
+///Ends the jump
+/datum/action/cooldown/spell/bunny_hop/proc/end_jump(mob/living/jumper)
+	jumper.remove_filter(BUNNY_HOP)
+	jumper.layer = initial(jumper.layer)
+	jumper.pass_flags = initial(jumper.pass_flags)
```

Comment:
Might want to cache their original pass flags so we're not clearing ones they already had from something else but _didn't_ have on round start
Pain in the ass that these can't have sources but there we are 😔

Reply:
Yes, you need to use `passtable_off` or you will reset stuff like dwarfism. https://github.com/tgstation/tgstation/pull/79698#issuecomment-1817666120